packages/@aws-cdk/aws-stepfunctions-tasks/lib/codebuild/start-build.ts

```diff
@@ -91,13 +121,56 @@ export class CodeBuildStartBuild extends sfn.TaskStateBase {
    * @internal
    */
   protected _renderTask(): any {
+    const sourceConfig = this.props.overrides?.source?.bind(this.props.project.stack, this.props.project);
+    const secondarySources = this.props.overrides?.secondarySources?.map(source => source.bind(this.props.project.stack, this.props.project));
     return {
       Resource: integrationResourceArn('codebuild', 'startBuild', this.integrationPattern),
       Parameters: sfn.FieldUtils.renderObject({
+        ArtifactsOverride: this.props.overrides?.artifacts?.bind(this.props.project.stack, this.props.project).artifactsProperty,
+        BuildspecOverride: this.props.overrides?.buildSpec?.toBuildSpec(),
+        BuildStatusConfigOverride: sourceConfig?.sourceProperty.buildStatusConfig,
+        // CacheOverride: , not supported yet
+        // CertificateOverride: , not supported yet
+        ComputeTypeOverride: this.props.overrides?.environment?.computeType,
+        DebugSessionEnabled: this.props.overrides?.debugSessionEnabled,
+        EncryptionKeyOverride: this.props.overrides?.encryptionKey?.keyArn,
+        EnvironmentTypeOverride: this.props.overrides?.environment?.buildImage?.type,
+        EnvironmentVariablesOverride: this.props.overrides?.environmentVariables ?
+          this.serializeEnvVariables(this.props.overrides?.environmentVariables!) :
+          (this.props.environmentVariablesOverride
+            ? this.serializeEnvVariables(this.props.environmentVariablesOverride)
+            : undefined),
+        GitCloneDepthOverride: sourceConfig?.sourceProperty.gitCloneDepth,
+        GitSubmodulesConfigOverride: sourceConfig?.sourceProperty.gitSubmodulesConfig,
+        IdempotencyToken: this.props.overrides?.idempotencyToken,
+        ImageOverride: this.props.overrides?.environment?.buildImage?.imageId,
+        ImagePullCredentialsTypeOverride: this.props.overrides?.environment?.buildImage?.imagePullPrincipalType,
+        InsecureSslOverride: sourceConfig?.sourceProperty.insecureSsl,
+        // LogsConfigOverride: , not supported yet
+        PrivilegedModeOverride: this.props.overrides?.environment?.privileged,
         ProjectName: this.props.project.projectName,
-        EnvironmentVariablesOverride: this.props.environmentVariablesOverride
-          ? this.serializeEnvVariables(this.props.environmentVariablesOverride)
-          : undefined,
+        // QueuedTimeoutInMinutesOverride: , not supported yet
```

Comment:
I wonder - can you bunch all of the override properties together, and put them at the end of this block? Not intermixed with the non-override properties.